data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -168,6 +164,24 @@ else if (isEventNameCreated(entry.getValue().get(0))) {
         return deleteMessageBatchRequestEntryCollection;
     }
 
+    private Optional<DeleteMessageBatchRequestEntry> processS3Object(
+            final Map.Entry<Message, List<S3EventNotification.S3EventNotificationRecord>> entry,
+            final S3ObjectReference s3ObjectReference) {
+        Optional<DeleteMessageBatchRequestEntry> deleteMessageBatchRequestEntry = Optional.empty();
+        // SQS messages won't be deleted if we are unable to process S3Objects because of S3Exception: Access Denied
+        try {
+            s3Service.addS3Object(s3ObjectReference);
+            deleteMessageBatchRequestEntry = Optional.of(buildDeleteMessageBatchRequestEntry(entry.getKey()));
+            sqsMessageDelayTimer.record(Duration.between(
+                    Instant.ofEpochMilli(entry.getValue().get(0).getEventTime().toInstant().getMillis()),
+                    Instant.now()
+            ));
```

Comment:
I want to point out that we don't delete SQS message if there is an exception in S3ObjectWorker. Let me know if you have any ideas on how to improve the user experience here.